### DISCUSSION TITLE
Unable to Connect Automatic1111 to Open WebUI

### DISCUSSION THREAD

**Discussion by NullEqualsZero** [open-webui/open-webui#7661] [Issues]
### Open WebUI
- **Installation Method:** Docker (running inside WSL)
- **Version:** `WEBUI_BUILD_VERSION c4ea31357f49d08a14c86b2bd85fdcd489512e91`
- **Python Version (in container):** `3.11.10`

### Automatic1111
- **Installation Method:** Local on Windows
- **Arguments Used:** `--listen --api`

## Environment

- **Ollama Version:** `0.4.7` (although not relevant) 
- **System Python Version (outside Docker):** `3.12.3`
- **Operating System:** 
  - **Base OS/WSL :** Windows 11/ WSL Ubuntu 
  - **Automatic1111 Host:** Windows (same as above)

## Confirmation:

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior

Automatic1111 should successfully connect to Open WebUI when the URL `http://127.0.0.1:7860/` is provided.

## Actual Behavior

When entering the URL `http://127.0.0.1:7860/` into Open WebUI, the following error is returned:
```
Oops! The URL you provided is invalid. Please double-check and try again.
```

The URL appears to be correct, so it’s unclear why this error is occurring.

### Docker Logs
The output of `docker logs open-webui` is attached in the file 
[logs.txt](https://github.com/user-attachments/files/18048980/logs.txt) 

### Browser Console Logs
Error messages from the browser console:
```
index.ts:73 GET http://127.0.0.1:8080/images/api/v1/config/url/verify 400 (Bad Request)
```

### Steps to Reproduce

1. Start Open WebUI in Docker with the following configuration:
   - Docker running inside WSL.
   - Open WebUI accessible via `http://127.0.0.1:8080`.
2. Start Automatic1111 locally on Windows with the arguments `--listen --api`.

(The whole file )

```bash
@echo off
set PYTHON=
set GIT=
set VENV_DIR=
set COMMANDLINE_ARGS=--xformers --listen --enable-insecure-extension-access --api
call webui.bat
```

3. Attempt to connect Automatic1111 to Open WebUI by providing the URL `http://127.0.0.1:7860/`.

### Observed Behavior

The error message is displayed:
```
Oops! The URL you provided is invalid. Please double-check and try again.
```

## Additional Information

- **Networking Context:** 
  - Docker runs inside WSL on the same machine where Automatic1111 is hosted on Windows.
  - Docker uses `127.0.0.1:8080` for Open WebUI.

If additional information or logs are required, please let me know. Thank you!

**Comment by ian-young** [comment]
If you are running Open WebUI in a Docker container but Automatic1111 on your host machine, you will need to reference to its URL as `http://host.docker.internal:7860` and not as `http://localhost:7860` as calling the latter is pointing the Docker container to its own `localhost` and not to the host's internal address.
